File: 997-find-the-town-judge/997-find-the-town-judge.js
/**
 * @param {number} N
 * @param {number[][]} trust
 * @return {number}
 */
var findJudge = function(n, trust) {
    const Trusted = new Array(n+1).fill(0);
    
    for(let [i,j] of trust) {
        Trusted[i] -= 1
        Trusted[j] += 1
    }

    for(let i = 1; i < Trusted.length; i++) {
        if ((n-1) === Trusted[i]) {
            return i;
        }
    }
    
    return -1
};

File: 997-find-the-town-judge/997-find-the-town-judge.java
class Solution {
    public int findJudge(int N, int[][] trust) {
        int[] Trusted = new int[N+1];
        for(int[] person : trust){
            Trusted[person[0]]--;
            Trusted[person[1]]++;
        }
        for(int i = 1;i < Trusted.length;i++){
            if(Trusted[i] == N-1) return i;
        }
        return -1;
    }
}